packages/admin-next/dashboard/src/v2-routes/campaigns/common/components/create-campaign-form-fields/create-campaign-form-fields.tsx

```diff
@@ -0,0 +1,268 @@
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}ends_at`}
+          render={({ field: { value, onChange, ref: _ref, ...field } }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("campaigns.fields.end_date")}</Form.Label>
+
+                <Form.Control>
+                  <DatePicker
+                    showTimePicker
+                    value={value ?? undefined}
+                    onChange={(v) => onChange(v ?? null)}
+                    {...field}
+                  />
+                </Form.Control>
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+      </div>
+
+      <div>
+        <Heading>{t("campaigns.budget.create.header")}</Heading>
+        <Text size="small" className="text-ui-fg-subtle">
+          {t("campaigns.budget.create.hint")}
+        </Text>
+      </div>
+
+      <Form.Field
+        control={form.control}
+        name={`${fieldScope}budget.type`}
+        render={({ field }) => {
+          return (
+            <Form.Item>
+              <Form.Label>{t("campaigns.budget.fields.type")}</Form.Label>
+
+              <Form.Control>
+                <RadioGroup
+                  className="flex gap-y-3"
+                  {...field}
+                  onValueChange={field.onChange}
+                >
+                  <RadioGroup.ChoiceBox
+                    className={clx("basis-1/2", {
+                      "border-2 border-ui-border-interactive":
+                        "spend" === field.value,
+                    })}
+                    value={"spend"}
+                    label={t("campaigns.budget.type.spend.title")}
+                    description={t("campaigns.budget.type.spend.description")}
+                  />
+
+                  <RadioGroup.ChoiceBox
+                    className={clx("basis-1/2", {
+                      "border-2 border-ui-border-interactive":
+                        "usage" === field.value,
+                    })}
+                    value={"usage"}
+                    label={t("campaigns.budget.type.usage.title")}
+                    description={t("campaigns.budget.type.usage.description")}
+                  />
+                </RadioGroup>
+              </Form.Control>
+              <Form.ErrorMessage />
+            </Form.Item>
+          )
+        }}
+      />
+
+      <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}budget.limit`}
+          render={({ field: { onChange, value, ...field } }) => {
+            return (
+              <Form.Item className="basis-1/2">
+                <Form.Label>{t("campaigns.budget.fields.limit")}</Form.Label>
+
+                <Form.Control>
+                  {isTypeSpend ? (
```

Comment:
suggestion: we should reset this value when we switch between usage limit and amount

Reply:
I tried this here:

```
useEffect(() => {
  form.setValue(`budget.limit`, undefined)
}, [watchValueType])
```

But its only resetting every 2nd click, any reccos on how to go about this?

Reply:
hmm seems that input is having problem with `undefined`, setting the value to `""` should do the trick: 
`form.setValue(`${fieldScope}budget.limit`, "")`

Reply:
nice that worked, moved it to the new branch.